test/e2e/e2e.go

```diff
@@ -0,0 +1,234 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	sourcesv1alpha1 "github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	servingV1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	rbacV1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	// Mysteriously required to support GCP auth (required by k8s libs).
+	// Apparently just importing it is enough. @_@ side effects @_@.
+	// https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	defaultNamespaceName = "e2etestfn3"
```

Comment:
This is the same namespace as eventing uses, is that on purpose?

Reply:
Not intentional, but the purpose of these namespaces is opaque to me with names like `e2etestfn3` and `e2etest`. Why are there two? @vaikas-google may be able to illuminate this.

Reply:
Original reason was that we should not have assumptions about the namespaces. Too many times I've seen things only being deployed to "default" and as soon as things get deployed to another namespace, things don't work. I think the original intent was to deploy the Receive Adapter into one namespace and function into another, but this model may not be supportable anymore.

Reply:
e2etestfn3 was where the function was to be deployed
e2etest where other bits were supposed to be deployed